osh/cmd_parse.py

```diff
@@ -1590,6 +1604,11 @@ def ParseOldCase(self, case_kw):
                     "This is a constant string.  You may want a variable like $x (parse_bare_word)",
                     loc.Word(w))
 
+        if w.tag() == word_e.Token:
+            token = cast(Token, w)
+            if token.id == Id.Op_Newline:
```

Comment:
I think we should disallow ANY `word_e.Token` -- we can consider renaming it to `word_e.Operator` to make that clear

An operator is a character like `() | ;` roughly  where `echo |` doesn't echo the character

This is in contrast to the CompoundWord.  The case pattern should be a CompoundWord


I tested this 

```
case |
in
esac
```

and it is similarly wrong in OSH


Similarly with `case ;` etc.

Please add one of those in `test/parse-errors.sh` too

Reply:
Good catch! I wouldn't be opposed to renaming `word_e.Token` -> `word_e.Operator` in another PR.